File: Java/LeetCode/Find the Distinct Difference Array.java
// https://leetcode.com/problems/find-the-distinct-difference-array

class Solution {

    public int[] distinctDifferenceArray(int[] nums) {
        int[] diff = new int[nums.length];

        for (int i = 0; i < nums.length; ++i) {
            Set<Integer> prefixNums = new HashSet<>();
            Set<Integer> suffixNums = new HashSet<>();
            
            for (int j = 0; j <= i; ++j) {
                prefixNums.add(nums[j]);
            }
            for (int j = i+1; j < nums.length; ++j) {
                suffixNums.add(nums[j]);
            }

            diff[i] = prefixNums.size() - suffixNums.size();
        }

        return diff;
    }

}

File: Kotlin/LeetCode/Find the Distinct Difference Array.kt
// https://leetcode.com/problems/find-the-distinct-difference-array

class Solution {

    fun distinctDifferenceArray(nums: IntArray): IntArray {
        var diff = IntArray(nums.size)

        for (i in 0..nums.size-1) {
            var prefixNums = mutableSetOf<Int>()
            var suffixNums = mutableSetOf<Int>()

            for (j in 0..i) {
                prefixNums.add(nums[j])
            }
            for (j in i+1..nums.size-1) {
                suffixNums.add(nums[j])
            }

            diff[i] = prefixNums.size - suffixNums.size
        }

        return diff
    }

}
